py/picca/io.py

```diff
@@ -349,14 +357,37 @@ def read_data(in_dir,
 
     # read data taking the mode into account
     blinding = "none"
-    if mode in ["desi_mocks","desi","desi_survey_tilebased", "spcframe", "spplate", "spec", "corrected-spec"]:
-        if mode in ["desi", 'desi_mocks']:
-            pix_data, is_mock = read_from_desi(in_dir, catalog, pk1d=pk1d)
-            if (not is_mock) and ('TILEID' in catalog.colnames) and np.any((catalog['TILEID']<60000)&(catalog['TILEID']>=1000)):
-                print("you are trying to run on DESI survey tiles!")
+    if mode in ["desi_mock","desi_healpix","desi","desi_survey_tilebased", "spcframe", "spplate", "spec", "corrected-spec"]:
+        if mode =='desi_mock': #I still don't think we need two different modes since we are checking if truth files exist...
+            desi_prefix = 'spectra'
+            desi_nside = 16
+            pix_data, is_mock = read_from_desi(in_dir, catalog, desi_prefix, desi_nside, pk1d=pk1d)
+            
+        elif mode == 'desi_healpix':
+            pix_data=[]
+            for survey_type in ['SV1','SV3','DESI'] :
+                desi_nside = 64
+                if survey_type == 'DESI':
+                    catalog_ = catalog[catalog['DESI_TARGET']>0]
```

Comment:
I'm using the columns "DESI_TARGET", "SV1_DESI_TARGET" and "SV3_DESI_TARGET" to infer the location of the spectra files. I think it would be clearer if the catalog had a column indicating from which survey we should read the spectra. In the current catalog there are no repited TARGETIDS, and I think there are targets from more than one survey mode, but there could be in general. What do you think? Also, if we request that column to be in the catalog, we can simplify some of this lines...

Reply:
We should ask about this to the data team, but I think there are "SURVEY" or "FLAVOUR" keywords somewhere. Otherwise, DESI_TARGET might already be enough. At any rate, this seems good enough for now.